prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -116,6 +117,26 @@ fun Bundle.addProvenanceReference() {
     }
 }
 
+/**
+ * Return true if Bundle contains an ELR in the MessageHeader.
+ *
+ * @return true if has a MesssageHeader that contains an R01 or ORU_R01, otherwise false.
+ */
+fun Bundle.isElr(): Boolean {
+    var isElr = false
+    if (this.type == Bundle.BundleType.MESSAGE && this.entry.isNotEmpty()) {
+        // By rule, the first entry must be a MessageHeader
```

Comment:
Can you link to documentation that says this? This makes me uneasy. See fun `Bundle.getDiagnosticReportNoObservations(): List<Base> {` below to see how we can utilize FHIRPath here for a better solution imo.

Reply:
Look at the documentation around "entry" for the [Bundle](https://www.hl7.org/fhir/bundle.html) resource

_This repeating element order: For bundles of type 'document' and 'message', the first resource is special (must be Composition or MessageHeader respectively). For all bundles, the meaning of the order of entries depends on the bundle type._

Reply:
I played around with FHIRPath a bit and I think it makes it a bit more readable and flat. I've also learned working here that specs are often not followed as closely as we anticipate.

```kotlin
val code = FhirPathUtils.evaluate(
        null,
        this,
        this,
        "Bundle.entry.resource.ofType(MessageHeader).event.code"
    )
        .filterIsInstance<CodeType>()
        .firstOrNull()
        ?.code
```